src/main/scala/monix/nio/tcp/AsyncTcpClientConsumer.scala

```diff
@@ -50,6 +59,7 @@ class AsyncTcpClientConsumer(host: String, port: Int) extends Consumer[Array[Byt
 
       override def onNext(elem: Array[Byte]): Future[Ack] = {
         val promise = Promise[Ack]()
+        connectedSignal.await(timeoutInSeconds, TimeUnit.SECONDS)
         client.writeChannel(ByteBuffer.wrap(elem), new Callback[Int] {
```

Comment:
I don't trust Java APIs to not throw exceptions, even when they take a callback with which it can report errors. This needs to be in a try/catch.